hibike/lib/hibike/hibike_device.cpp

```diff
@@ -31,47 +31,63 @@ void hibike_loop() {
     if (read_message(&hibikeBuff) == -1) {
       toggleLED();
     } else {
+      int offset;
       switch (hibikeBuff.messageID) {
 
         case SUBSCRIPTION_REQUEST:
           // change subDelay and send SUB_RESP
-          subDelay = *((uint16_t*) &hibikeBuff.payload[0]);
-          send_subscription_response(&UID, subDelay);
+          subDelay = *((uint16_t*) &hibikeBuff.payload[2]);
+          params = *((uint16_t*) &hibikeBuff.payload[0]);
+          send_subscription_response(params, subDelay, &UID);
           break;
 
         case SUBSCRIPTION_RESPONSE:
           // Unsupported packet
           toggleLED();
           break;
 
-        case DATA_UPDATE:
-          // Unsupported packet
-          toggleLED();
+        case DEVICE_WRITE:
+          //loop over params
+          old_params = params;
+          offset = 2;
+          params = *((uint16_t*)&hibikeBuff.payload[0]);
+          for (uint16_t count = 0; (params >> count) > 0; count++) {
+            if (params & (1<<count)){
+              int status = device_write(count, &hibikeBuff.payload[offset], hibikeBuff.payload_length-offset);
+              if(status){
+                offset += status;}
+              else{
+                params = params & ~(1<<count);}
+              }
+          }
+
+          // params = hibikeBuff.payload[0] - 1;
+          // value = *((uint32_t*) &hibikeBuff.payload[DEVICE_PARAM_BYTES]);
+          //write values
+          *((uint16_t*)&hibikeBuff.payload[0]) = params;
+          send_data_update(*((uint16_t*) &hibikeBuff.payload[0]));
+          params = old_params;
           break;
 
-        case DEVICE_UPDATE:
-          param = hibikeBuff.payload[0] - 1;
-          value = *((uint32_t*) &hibikeBuff.payload[DEVICE_PARAM_BYTES]);
-          send_device_response(param + 1, device_update(param, value));
-          break;
+        case DEVICE_READ:
 
-        case DEVICE_STATUS:
-          param = hibikeBuff.payload[0] - 1;
-          send_device_response(param + 1, device_status(param));
+          send_data_update(*((uint16_t*) &hibikeBuff.payload[0]));
+          // param = hibikeBuff.payload[0] - 1;
+          // send_device_response(param + 1, device_status(param));
           break;
 
-        case DEVICE_RESPONSE:
+        case DEVICE_DATA:
           // Unsupported packet
           toggleLED();
           break;
 
-        case PING_:
-          send_subscription_response(&UID, subDelay);
+        case PING:
+          send_subscription_response(params, subDelay, &UID);
           break;
 
-        case DESCRIPTION_REQUEST:
-          send_description_response(DESCRIPTION);
-          break;
+        // case DESCRIPTION_REQUEST:
```

Comment:
Why is this one commented out, but the other ones are now commented out but have a comment `Unsupported packet`?